### PR TITLE
Convert errorAndCallback ternary to arrow function

### PR DESCRIPTION
**Pull request by athomann**
**What kind of change does this PR introduce?**
Refactoring

**Did you add tests for your changes?**
Minor refactor. Not applicable?

**Summary**
Convert function to arrow function, removing the need to bind.

**Does this PR introduce a breaking change?**
no

### CODE REVIEW

**Comment by athomann** [comment]
If core team would like to move in this direction, I'd be happy to refactor instances of `bind(this)` to arrow functions throughout the codebase.

File: lib/Compilation.js [hunk 1]
```diff
@@ -362,13 +362,13 @@ class Compilation extends Tapable {
 	_addModuleChain(context, dependency, onModule, callback) {
 		const start = this.profile && Date.now();
 
-		const errorAndCallback = this.bail ? function errorAndCallback(err) {
+		const errorAndCallback = this.bail ? (err) => {
 			callback(err);
-		} : function errorAndCallback(err) {
```
Comment (TheLarkInn) [inline lib/Compilation.js]: Looking back on the blame for this I believe the reasoning was being a named function. Could we preserve the name for debugging and traces through outer scope assignment etc?

Comment (athomann) [inline lib/Compilation.js]: Are you referring so it stays in the stack trace?  Did a little digging and it looks like it shows up when assigned to a variable? 

http://lukecod.es/2015/12/01/es6-arrow-function-stack-traces/

https://stackoverflow.com/questions/27977525/how-do-i-write-a-named-arrow-function-in-es2015#answers-header

Comment (TheLarkInn) [inline lib/Compilation.js]: Oh bahh I totally missed the initial assignment. Yes this makes sense and I'm down with that.

**Comment by TheLarkInn** [comment]
@sokra I'll let you merge once you jump on in a few.

**Pull request merged**